smart_open/ssh.py

```diff
@@ -43,18 +44,22 @@
 DEFAULT_PORT = 22
 
 
-def _connect(hostname, username, port):
+def _connect(hostname, username, port, rsakey = None):
     key = (hostname, username)
     ssh = _SSH.get(key)
     if ssh is None:
         ssh = _SSH[key] = paramiko.client.SSHClient()
         ssh.load_system_host_keys()
         ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
-        ssh.connect(hostname, port, username)
+        if rsakey:
```

Comment:
I think the variable naming here is a bit confusing. How is rsakey different from sshkey and pkey?

Also, is there a benefit to passing rsakey instead of pkey?

Furthermore, I think instead of passing just the key, we should pass connect_kwargs as dictionary and then pass them all to the connect function. That will allow us to support other parameters on top of pkey (there are many: https://docs.paramiko.org/en/2.4/api/client.html#paramiko.client.SSHClient.connect)